test/physics/em/Rayleigh.test.cc

```diff
@@ -33,46 +46,114 @@ class RayleighInteractorTest : public celeritas_test::InteractorHostTestBase
     }
 
     void sanity_check(const Interaction& interaction) const
     {
         ASSERT_TRUE(interaction);
 
         // Check change to parent track
-        EXPECT_GT(this->particle_track().energy().value(),
+        EXPECT_EQ(this->particle_track().energy().value(),
                   interaction.energy.value());
-        EXPECT_LT(0, interaction.energy.value());
-        EXPECT_SOFT_EQ(1.0, celeritas::norm(interaction.direction));
+        EXPECT_EQ(1.0, interaction.energy.value());
         EXPECT_EQ(celeritas::Action::scattered, interaction.action);
-
-        // XXX Check secondaries
-
-        // Check conservation between primary and secondaries
-        this->check_conservation(interaction);
     }
 
   protected:
-    celeritas::detail::RayleighInteractorPointers pointers_;
+    std::shared_ptr<RayleighModel>            model_;
+    celeritas::detail::RayleighNativePointers pointers_;
 };
 
 //---------------------------------------------------------------------------//
 // TESTS
 //---------------------------------------------------------------------------//
 
-TEST_F(RayleighInteractorTest, basic) {}
+TEST_F(RayleighInteractorTest, basic)
+{
+    const int num_samples = 4;
+
+    // Sample an element (TODO: add ElementSelector)
+    ElementId el_id{0};
+
+    // Create the interactor
+    RayleighInteractor interact(this->model_->host_pointers(),
+                                this->particle_track(),
+                                this->direction(),
+                                el_id);
+
+    RandomEngine& rng_engine = this->rng();
+
+    // Produce four samples from the original/incident photon
+    std::vector<double> angle;
+
+    for (CELER_MAYBE_UNUSED auto i : celeritas::range(num_samples))
+    {
+        celeritas::Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+        this->sanity_check(result);
+
+        angle.push_back(dot_product(result.direction, this->direction()));
+    }
+
+    const double expected_angle[] = {-0.0798080588007317,
+                                     -0.199218754040088,
+                                     -0.420659925857054,
+                                     0.464669752940999};
+
+    EXPECT_VEC_SOFT_EQ(expected_angle, angle);
+}
+
+TEST_F(RayleighInteractorTest, stress_test)
+{
+    const int num_samples = 8192;
+
+    // Sample an element
+    ElementId el_id{0};
+
+    // Create the interactor
+    RayleighInteractor interact(this->model_->host_pointers(),
+                                this->particle_track(),
+                                this->direction(),
+                                el_id);
+
+    RandomEngine& rng_engine = this->rng();
+
+    // Produce four samples from the original/incident photon
+    for (CELER_MAYBE_UNUSED auto i : celeritas::range(num_samples))
+    {
+        celeritas::Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+        this->sanity_check(result);
+    }
```

Comment:
Can you extend this test function so it's over a handful of energy values and test `rng.count()` to test how the rejection rate changes over the energy range of the model?

Reply:
Extended test with multiple energy points and added rng.count().